File: Java/Triangolo/Triangolo.java
public class Triangolo {
    private Punto a;
    private Punto b;
    private Punto c;

    public Triangolo(Punto a, Punto b, Punto c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    //setter
    public void setA(Punto a) {
        this.a = a;
    }
    public void setB(Punto b) {
        this.b = b;
    }
    public void setC(Punto c) {
        this.c = c;
    }
    //getter
    public Punto getA() {
        return a;
    }
    public Punto getB() {
        return b;
    }
    public Punto getC() {
        return c;
    }
    
    //metodi

    /**
     * calcola il perimetro del triangolo
     * @return (double) perimetro
     */
    public double perimetro() {
        return a.distanza(b) + b.distanza(c) + c.distanza(a);
    }

    /**
     * calcola l'area del triangolo
     * @return (double) area
     */
    public double area() {
        return Math.sqrt((perimetro()/2) * ((perimetro()/2) - a.distanza(b)) * ((perimetro()/2) - b.distanza(c)) * ((perimetro()/2) - c.distanza(a)));
    }

    /**
     * restituisce una stringa che descrive il triangolo
     */
    public String toString() {
        return ("Punto A: " + a.toString() + ", Punto B: " + b.toString() + ", Punto C: " + c.toString() + String.format("Perimetro: %.2f e Area: %.2f", perimetro(), area()));
    }

    /**
     * controlla se due triangoli sono uguali
     * @param (Triangolo t) 
     * @return (boolean) true se sono uguali, false se non sono uguali
     */
    public boolean equals(Triangolo t) {
        if (a.equals(this.a, t.getA()) && b.equals(this.b, t.getB()) && c.equals(this.c, t.getC())) {
            return true;
        }
        return false;
    }

    /**
     * controlla se due triangoli hanno la stessa area
     * @param (Triangolo t)
     * @return (boolean) true se hanno la stessa area, false se non hanno la stessa area
     */
    public boolean equivale(Triangolo t) {
        if (this.area() == t.area())
            return true;
        return false;
    }
}

File: Python/Triangolo/Triangolo.py
import math

class Triangolo:
    def __init__(self,a,b,c):
        self.a = a
        self.b = b
        self.c = c
        
    def __str__(self):
        return f"A ({self.a.getX()},{self.a.getY()}), B ({self.b.getX()},{self.b.getY()}), C ({self.c.getX()}, {self.c.getY()})"
    
    #getter
    def getA(self):
        return self.a
    def getB(self):
        return self.b
    def getC(self):
        return self.c

    #setter
    def setA(self, x, y):
        self.a.setX(x)
        self.a.setY(y)
        
    def setB(self, x, y):
        self.b.setX(x)
        self.b.setY(y)
        
    def setC(self, x, y):
        self.c.setX(x)
        self.c.setY(y)
        
    #methods
    def perimetro(self):
        return (self.getA().distanza(self.b) + self.getB().distanza(self.c) + self.getC().distanza(self.a))
    
    def area(self):
        return math.sqrt(((self.perimetro() / 2) - self.getA().distanza(self.getB())) * 
                         ((self.perimetro() / 2) - self.getB().distanza(self.getC())) * 
                         ((self.perimetro() / 2) - self.getC().distanza(self.getA())))
                         
        